apps/api/src/app/workflows-v2/usecases/create-variables-object/create-variables-object.usecase.ts

```
@@ -24,7 +39,35 @@ export class CreateVariablesObject {
     ];
     const showIfVariables = this.extractMailyAttribute(controlValues, MailyAttrsEnum.SHOW_IF_KEY);
 
-    return keysToObject(variables, arrayVariables, showIfVariables);
+    const variablesObject = keysToObject(variables, arrayVariables, showIfVariables);
+    if (isEnhancedDigestEnabled) {
+      return this.ensureEventsArray(variablesObject);
+    }
+
+    return variablesObject;
+  }
+
+  private ensureEventsArray(variablesObject: Record<string, unknown>) {
+    const stepsObject = (variablesObject.steps as Record<string, unknown>) ?? {};
+
+    Object.keys(stepsObject)
+      .filter((stepId) => typeof stepsObject[stepId] === 'object')
```

Is it possible to have an empty object at this point? If not let's remove this extra check so that it fails fast if it happens.

---

it shouldn't be the case, yes let's do that 🙌